src/cast_core/test/CrudActorYamlTests.yml

```diff
@@ -716,3 +716,285 @@ Tests:
+          OperationCommand:
+            Filter: { a: 1 }
+            Options:
+              Tailable: true
+              AwaitData: true
+    ExpectAllEvents:
+      cursorType: tailable_await
+
+  - Description: Test that specifying 'awaitData' without 'tailable' results in an error.
+    Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: { a: 1 }
+            Options:
+              AwaitData: true
+    Error: | 
+      .*Cannot set 'awaitData' to true without also setting 'tailable' to true.*
+
+  - Description: Test that an error is thrown when an invalid projection is specified in 'findOne'.
+    Operations:
+      - OperationName: findOne
+        OperationCommand:
+          Filter: {a: 100}
+          Options:
+            Projection: {_id: 0, a: 0, b: 1}
+    Error: |
+      .*Cannot do inclusion on field b in exclusion projection.*
+
+  - Description: Test that a 'findOne' command applies the specified projection.
+    Operations:
+      - OperationName: findOne
+        OperationCommand:
+          Filter: {a: 100}
+          Options:
+            Projection: {_id: 0, b: 1}
+    ExpectAllEvents:
+      projection: {_id: 0, b: 1}
+
+  - Description: Test that a 'findOne' command applies the specified sort.
+    Operations:
+      - OperationName: findOne
+        OperationCommand:
+          Filter: { a: 1 }
+          Options:
+            Sort: { a: -1 }
+    ExpectAllEvents:
+      sort: { a: -1 }
+
+  - Description: Test that a 'findOne' command applies the specified collation.
+    Operations:
+      - OperationName: findOne
+        OperationCommand:
+          Filter: { a: 1 }
+          Options:
+            Collation: { locale: "simple" }
+    ExpectAllEvents:
+      collation: { locale: "simple" }
+
+  - Description: Test that a 'findOne' command applies a hint specified as a string.
+    Operations:
+      - OperationName: findOne
+        OperationCommand:
+          Filter: { a: 1 }
+          Options:
+            Hint: "_id_1"
+    ExpectAllEvents:
+      hint: "_id_1"
+
+  - Description: Test that a 'findOne' command uses the specified comment.
+    Operations:
+      - OperationName: findOne
+        OperationCommand:
+          Filter: { a: 1 }
+          Options:
+            Comment: &comment "Please show up in the logs"
+    ExpectAllEvents:
+      comment: *comment
+
+  - Description: Test that a 'findOne' command ignores the specified limit.
+    Operations:
+      - OperationName: findOne
+        OperationCommand:
+          Filter: { a: 1 }
+          Options:
+            Limit: 20000
+    ExpectAllEvents:
+      limit: 1
+
+  - Description: Test that a 'findOne' command applies the specified skip.
+    Operations:
+      - OperationName: findOne
+        OperationCommand:
+          Filter: { a: 1 }
+          Options:
+            Skip: 2
+    ExpectAllEvents:
+      skip: 2
+
+  - Description: Test that a 'findOne' command applies the specified batchSize.
```

Comment:
Also wacky. I'd argue that `batchSize` should be explicitly rejected when used with the findOne actor, implemented with a special check in `CrudActor.cpp`.